### PR TITLE
PMM-7847 API for MySQL TLS.

### PR DESCRIPTION
**Pull request by JiriCtvrtka**

### CODE REVIEW

File: connectionchecker/connection_checker.go [hunk 1]
```diff
@@ -89,17 +90,30 @@ func (cc *ConnectionChecker) sqlPing(ctx context.Context, db *sql.DB) error {
 	return err
 }
 
-func (cc *ConnectionChecker) checkMySQLConnection(ctx context.Context, dsn string) *agentpb.CheckConnectionResponse {
+func (cc *ConnectionChecker) checkMySQLConnection(ctx context.Context, dsn string, files *agentpb.TextFiles) *agentpb.CheckConnectionResponse {
 	var res agentpb.CheckConnectionResponse
+	var err error
 
-	cfg, err := mysql.ParseDSN(dsn)
+	if strings.Contains(dsn, "tls=custom") {
```
Comment (percona-csalguero) [inline connectionchecker/connection_checker.go:97]: For now, leave it as it is, but what if I set the password to `tls=custom`?
We need a better wat to detect if TLS is enabled.

File: tlshelpers/mysql.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+	ca := x509.NewCertPool()
+	cert, err := tls.X509KeyPair([]byte(files["tlsCert"]), []byte(files["tlsKey"]))
+	if err != nil {
+		return errors.Wrap(err, "register MySQL client cert failed")
+	}
+
+	if ok := ca.AppendCertsFromPEM([]byte(files["tlsCa"])); ok {
+		err = mysql.RegisterTLSConfig("custom", &tls.Config{
+			InsecureSkipVerify: false,
```
Comment (percona-csalguero) [inline tlshelpers/mysql.go:47]: If this is always false, it would be hard to test with selft signed certs

**Review by percona-csalguero — APPROVED** [review]

File: connectionchecker/connection_checker.go (hunk 1 above)
Comment (JiriCtvrtka) [reply connectionchecker/connection_checker.go]: Fixed.

File: tlshelpers/mysql.go (hunk 2 above)
Comment (JiriCtvrtka) [reply tlshelpers/mysql.go]: Yes, thank you.

File: tlshelpers/mysql.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+		if err != nil {
+			return errors.Wrap(err, "register MySQL CA cert failed")
+		}
+	}
+
+	return nil
+}
+
+// ProcessMySQLCertsArgs generate right args for given certificates.
+func ProcessMySQLCertsArgs(process *process.Params, files map[string]string, tempDir string, tlsSkipVerify bool) func() {
```
Comment (BupycHuk) [inline tlshelpers/mysql.go:60]: this should be done on pmm-managed side.

File: agents/supervisor/supervisor.go [hunk 4]
```diff
@@ -342,7 +343,25 @@ func (s *Supervisor) startProcess(agentID string, agentProcess *agentpb.SetState
...
+				return err
+			}
+		}
+
+		tempDir := filepath.Join(s.paths.TempDir, strings.ToLower(agentType), agentID)
+		cleanCerts = tlshelpers.ProcessMySQLCertsArgs(processParams, agentProcess.GetTextFiles(), tempDir, agentProcess.TlsSkipVerify)
+	default:
+	}
+
 	process := process.New(processParams, agentProcess.RedactWords, l)
```
Comment (BupycHuk) [inline agents/supervisor/supervisor.go:363]: all this block looks not required.

**Review by BupycHuk — CHANGES_REQUESTED** [review]

File: tlshelpers/mysql.go (hunk 3 above)
Comment (JiriCtvrtka) [reply tlshelpers/mysql.go]: Moved to pmm-managed.

**Review by BupycHuk — APPROVED** [review]
Approving it, let's create task for refactoring

**Comment by JiriCtvrtka** [comment]
Task for refactoring: https://jira.percona.com/browse/PMM-8014

**Pull request merged**